### ISSUE TITLE
#12253 FIX: Handle utf-8 output by kpathsea on Windows -- possibly causing issues

### ISSUE THREAD

**Issue by pharshalp** [matplotlib/matplotlib#12308]
### Bug report
I like to test my code against the bleeding edge version of matplotlib (by installing it in a conda virtual environment) on my work machine (Windows PC). Since I do not have admin rights to install the necessary tools to build from source, I download the wheels from appveyor and use pip install.

The following code snippet (representing a part of my working code) started giving an error after I installed "matplotlib-3.0.0rc1+424.g19420b97.dirty-cp36-cp36m-win_amd64.whl" (which corresponds to Merge pull request #12253 from anntzer/kpathsea-encoding FIX: Handle utf-8 output by kpathsea on Windows). The wheel which was built for the commit before that, matplotlib-3.0.0rc1+420.gb2c89917.dirty-cp36-cp36m-win_amd64.whl works fine (I reinstalled it and the code snippet works).

I am not sure if this is the right way to test the bleeding edge code (when you do not have admin rights to install the tools necessary to build from source). If the procedure I am following is not correct, please feel free to close the issue.

**Code for reproduction**
```python
import matplotlib.pyplot as plt
import numpy as np
from matplotlib import rc
rc('text', usetex=True)
rc('text.latex', preamble=r'\usepackage{fourier}, \usepackage[T1]{fontenc}')
fig, ax = plt.subplots()
ax.plot(np.arange(10))
fig.savefig('test.pdf')
```

**Actual outcome**

```
OSError: [Errno 22] Invalid argument: 'C:/Program Files/MiKTeX 2.9/fonts/tfm/jknappen/ec/ecss1000.tfm
\r'
```

**Expected outcome**

**Matplotlib version**

  * Operating system: Windows 10
  * Matplotlib version: 3.0.0rc1+424.g19420b97.dirty
  * Matplotlib backend (`print(matplotlib.get_backend())`): Qt5Agg

  * Python version:
  * Jupyter version (if applicable):
  * Other libraries:

**Comment by anntzer** [comment]
Sorry about that.  I think the following patch should suffice, can you confirm it?
```
diff --git i/lib/matplotlib/dviread.py w/lib/matplotlib/dviread.py
index f064113ad..a47320176 100644
--- i/lib/matplotlib/dviread.py
+++ w/lib/matplotlib/dviread.py
@@ -1017,16 +1017,16 @@ def find_tex_file(filename, format=None):
     if format is not None:
         cmd += ['--format=' + format]
     cmd += [filename]
-    try:  # Below: strip final newline.
-        result = cbook._check_and_log_subprocess(cmd, _log)[:-1]
+    try:
+        result = cbook._check_and_log_subprocess(cmd, _log)
     except RuntimeError:
         return ''
     if os.name == 'nt':
         # On Windows only, kpathsea appears to use utf-8 output(?); see
         # __win32_fputs in the kpathsea sources and mpl issue #11848.
-        return result.decode('utf-8')
+        return result.decode('utf-8')[:-2]  # Strip final '\r\n'.
     else:
-        return os.fsdecode(result)
+        return os.fsdecode(result)[:-1]  # Strip final '\n'.
 
 
 @lru_cache()
```

We should really reenable tex testing on appveyor...

**Comment by pharshalp** [comment]
Thanks! the proposed fix works.

**Comment by ImportanceOfBeingErnest** [comment]
Reopening. While it may be enough of a fix for you @pharshalp we would like to let the rest of the world also be able to use tex within matplotlib on windows, won't we? ;-) 
This should then be automatically closed once #12356 (or any other fix) is merged.

**Comment by pharshalp** [comment]
sure. I didn't take a look at the PR and thought it was already merged. Just wanted to bring down the count of the "open issues" :D and didn't know that the issue automatically gets closed once the PR is merged.

**Comment by pharshalp** [comment]
closed by #12351
